framework/source/class/qx/data/controller/List.js

```diff
@@ -409,7 +415,12 @@ qx.Class.define("qx.data.controller.List",
       qx.ui.core.queue.Widget.add(this);
 
       // update on filtered lists... (bindings need to be renewed)
-      this.update();
+      if (!this.__warningIssued) {
+        this.warn(this.classname + " can cause item model properties to change expectedly; please consider using qx.data.controller.ListOfObjects as a soltions.  See https://github.com/qooxdoo/qooxdoo/issues/196 for more details");
+        this.__warningIssued = true;
+      }
+
+      qx.event.Timer.once(this.update, this, 0);
```

Comment:
Although that simple one-liner is a bit of a kludge, maybe we should just
document why that's been put there (with a link to the discussion), and
*not* add or deprecate a class? I recall from earlier in the discussion
that there was stuff going on that really wasn't needed (adding and then
removing a listener, I think) but retaining a long-standing API has pretty
high value...

Derrell


On Fri, Apr 1, 2016 at 12:47 PM Tobias Oetiker <notifications@github.com>
wrote:

> In framework/source/class/qx/data/controller/List.js
> <https://github.com/qooxdoo/qooxdoo/pull/197#discussion_r58233129>:
>
> > @@ -409,7 +415,12 @@ qx.Class.define("qx.data.controller.List",
> >        qx.ui.core.queue.Widget.add(this);
> >
> >        // update on filtered lists... (bindings need to be renewed)
> > -      this.update();
> > +      if (!this.__warningIssued) {
> > +        this.warn(this.classname + " can cause item model properties to change expectedly; please consider using qx.data.controller.ListOfObjects as a soltions.  See https://github.com/qooxdoo/qooxdoo/issues/196
>
>  for more details");
> > +        this.__warningIssued = true;
> > +      }
> > +
> > +      qx.event.Timer.once(this.update, this, 0);
>
> this will actually fix the problem seen in the original bugreport, right ?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/qooxdoo/qooxdoo/pull/197/files/7aefcc6e3b8348bedd58bc989bca01cba20bae02#r58233129>
>